lib/ansible/modules/cloud/docker/docker_login.py

```diff
@@ -130,22 +131,146 @@
 import os
 import re
 import traceback
+from ansible.module_utils._text import to_bytes, to_text
 
 try:
     from docker.errors import DockerException
+    from docker import auth
+
+    # Earlier versions of docker/docker-py put decode_auth
+    # in docker.auth.auth instead of docker.auth
+    if hasattr(auth, 'decode_auth'):
+        from docker.auth import decode_auth
+    else:
+        from docker.auth.auth import decode_auth
+
 except ImportError:
     # missing Docker SDK for Python handled in ansible.module_utils.docker.common
     pass
 
-from ansible.module_utils._text import to_bytes, to_text
 from ansible.module_utils.docker.common import (
     AnsibleDockerClient,
+    HAS_DOCKER_PY,
     DEFAULT_DOCKER_REGISTRY,
     DockerBaseClass,
     EMAIL_REGEX,
     RequestException,
 )
 
+NEEDS_DOCKER_PYCREDS = False
+
+# Early versions of docker/docker-py rely on docker-pycreds for
+# the credential store api.
+if HAS_DOCKER_PY:
+    try:
+        from docker.credentials.errors import StoreError, CredentialsNotFound
+        from docker.credentials import Store
+    except ImportError:
+        try:
+            from dockerpycreds.errors import StoreError, CredentialsNotFound
+            from dockerpycreds.store import Store
+        except ImportError as exc:
+            HAS_DOCKER_ERRROR = str(exc)
+            NEEDS_DOCKER_PYCREDS = True
+
+
+if NEEDS_DOCKER_PYCREDS:
```

Comment:
What happens if it is missing? Will the module still work? (Right now, it never checks this variable again.)

Reply:
Right now, it won't fail with `ImportError` because you're catching that. You don't need to copy any messages from `module_utils`, you need to add specific messages about the imports which fail only for this module. So there won't be any duplication.

The commit you cite (docker/docker-py@545c5ac3cca04fbb95622f2440e15af186ff19fd) first appeared in docker-py 1.10.0. Does it mean that from 1.10.0 no additional requirements are needed (since the code is either included in docker-py itself, or installing docker-py requires te necessary module)? (And for 1.8.x and 1.9.x, they are needed?)